owa/app/js/components/manage_patient_identifier_sources/existingIdentifierSources.jsx

```diff
@@ -0,0 +1,95 @@
+import React, {Component} from 'react';
+import {Table, Button, Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
+import ViewSource from './viewSource';
+import ConfigureSource from './configureSource';
+
+export default class ExistingIdentifierSources extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      modal: false,
+      modalContent: '',
+      identifierSources: [{
+        identifierType: ['OpenMRS ID'],
+        sourceType: ['Local Identifier Generator'],
+        sourceName: ['New Source']
+      }]
+    };
+
+    this.toggle = this.toggle.bind(this);
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  //show the modal
+  toggle() {
+    this.setState({
+      modal: !this.state.modal
```

Comment:
Why do you choose to handle the modal display this way?

Reply:
The modal component has a prop isOpen which when true the modal opens and when false the modal closes. This is an efficient way to open and close the modal. What are your thoughts?